roles/kubernetes/preinstall/tasks/etchosts.yml

```diff
@@ -17,7 +17,7 @@
     line: "{{ loadbalancer_apiserver.address }} {{ apiserver_loadbalancer_domain_name| default('lb-apiserver.kubernetes.local') }}"
     state: present
     backup: yes
-  when: loadbalancer_apiserver is defined and apiserver_loadbalancer_domain_name is defined
+  when: loadbalancer_apiserver.address is defined and apiserver_loadbalancer_domain_name is defined
```

Comment:
Can you be more specific? As of my understanding this should fail for Loadbalancers where no static IP can be supplied (e.g. AWS Elastic Loadbalancer) hence no entry is necessary in etchosts. 
Two possible cases:

1.  API Loadbalancer has an static IP --> entry in etchosts is necessary --> test doesnt fail
 loadbalancer_apiserver:
   address: 10.250.xxx.xxx
   port: 443

2. API Loadbalancer doesnt has  an static IP --> entry in etchosts is not required ( e.g. AWS Elastic Loadbalancer) --> Test should fail
 loadbalancer_apiserver:
    port: 443